File: lib/src/models/log_entry.dart
/// Log level for RiviumTrace logging
enum LogLevel {
  trace('trace'),
  debug('debug'),
  info('info'),
  warn('warn'),
  error('error'),
  fatal('fatal');

  final String value;
  const LogLevel(this.value);
}

/// A single log entry to be sent to RiviumTrace
class LogEntry {
  final String message;
  final LogLevel level;
  final DateTime? timestamp;
  final Map<String, dynamic>? metadata;
  final String? userId;

  LogEntry({
    required this.message,
    this.level = LogLevel.info,
    this.timestamp,
    this.metadata,
    this.userId,
  });

  Map<String, dynamic> toJson() {
    return {
      'message': message,
      'level': level.value,
      if (timestamp != null) 'timestamp': timestamp!.toIso8601String(),
      if (metadata != null) 'metadata': metadata,
      if (userId != null) 'userId': userId,
    };
  }
}

File: android/src/main/kotlin/co/rivium/trace/sdk/models/LogEntry.kt
package co.rivium.trace.sdk.models

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.TimeZone

/**
 * Log level for RiviumTrace logging
 */
enum class LogLevel(val value: String) {
    TRACE("trace"),
    DEBUG("debug"),
    INFO("info"),
    WARN("warn"),
    ERROR("error"),
    FATAL("fatal")
}

/**
 * A single log entry to be sent to RiviumTrace
 */
data class LogEntry(
    val message: String,
    val level: LogLevel = LogLevel.INFO,
    val timestamp: Date = Date(),
    val metadata: Map<String, Any?>? = null,
    val userId: String? = null
) {
    fun toMap(): Map<String, Any?> {
        val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US).apply {
            timeZone = TimeZone.getTimeZone("UTC")
        }

        return buildMap {
            put("message", message)
            put("level", level.value)
            put("timestamp", dateFormat.format(timestamp))
            metadata?.let { put("metadata", it) }
            userId?.let { put("userId", it) }
        }
    }
}
